tests/phpunit/tests/user/author.php

```diff
@@ -144,4 +145,42 @@ public function test_get_the_author_posts_link_with_permalinks() {
 		unset( $GLOBALS['authordata'] );
 	}
 
+	/**
+	 * @ticket 51859
+	 */
+	public function test_get_author_url_link() {
+		$author_url          = get_the_author_meta( 'url' );
+		$author_display_name = get_the_author();
+
+		$author_url_link_html = sprintf(
+			'<a href="%1$s" title="%2$s" rel="author external">%3$s</a>',
+			esc_url( $author_url ),
+			esc_attr( sprintf( 'Visit %s&#8217;s website', $author_display_name ) ),
+			$author_display_name
+		);
+
+		$this->assertSame( get_the_author_link(), $author_url_link_html );
+	}
+
+	public function filter_author_url_link_html( $link, $url, $author_data ) {
+		return '<a href="' . $url . '">' . $author_data->display_name . '</a>';
+	}
+
+	/**
+	 * @ticket 51859
+	 */
+	public function test_filtered_get_author_url_link() {
+		$author_url          = get_the_author_meta( 'url' );
+		$author_display_name = get_the_author();
+
+		$expected_filtered_author_url_link_html = sprintf(
+			'<a href="%1$s">%2$s</a>',
+			esc_url( $author_url ),
+			$author_display_name
+		);
+
+		add_filter( 'author_url_link_html', array( $this, 'filter_author_url_link_html' ), 10, 3 );
+
+		$this->assertSame( get_the_author_link(), $expected_filtered_author_url_link_html );
```

Comment:
Last week I found out the test suite has a helper for ensuring a filter has run.

```suggestion
		$filter = new MockAction();
		add_filter( 'author_url_link_html', array( &$filter, 'filter' ) );
		get_the_author_link();
		$this->assertSame( 1, $filter->get_call_count() );
		$this->assertSame( array( 'author_url_link_html' ), $a->get_tags() );
```

You'll also need to remove `filter_author_url_link_html`.

Reply:
@peterwilsoncc So, it's enough that the filter is just "run" then?  We don't need to validate that a filtered value is what we would expect in a "happy path" test?  Sorry for my ignorance, as this is my first time seeing this pattern in Core.

Reply:
I think it's enough to test that the filter runs.

You could test that `$this->assertSame( $unmodified_link, $filter->get_args()[0][0] )` if you wish to confirm the correct data is passed to the filter. That will ensure future code changes don't break the passed data.

Reply:
This seems like a good approach. We could probably use this pattern more often in other tests, where the only purpose is to make sure the result is filterable.